src/orion/core/worker/warm_start/serializable.py

```diff
@@ -0,0 +1,148 @@
+import inspect
+from dataclasses import asdict, fields, is_dataclass
+from typing import (
+    Any,
+    Dict,
+    ItemsView,
+    Iterator,
+    KeysView,
+    Mapping,
+)
+from collections.abc import ItemsView
+from logging import getLogger as get_logger
+from orion.core.utils.flatten import flatten, unflatten
+
+logger = get_logger(__name__)
+
+
+class SerializableMixin(Mapping[str, Any]):
```

Comment:
Is this coming from another package your are working on? I'm wondering if it would be best to add dependencies instead of maintaining+testing it.

Reply:
This comes from simple-parsing. I doubt that you'd want to add a dependency to it, unless that's necessary.
(Although, simple-parsing is a really lightweight package.)

Reply:
There are other (perhaps better) alternatives though (third-party libraries such as Pydantic etc). I'll take a look.

Reply:
I don't know very well pydantic. If you think that is a good alternative than we could use it. 

Otherwise, do you think simple-parsing could help improve the cmdline parsers of Oríon? (thinking of commands like `hunt`, `status`, `db`, etc).

Reply:
Removed in 1b29a3813db57aa3f12e09cc49ca335fb5b15f4c